grails-app/controllers/org/bbop/apollo/OrganismController.groovy

```diff
@@ -123,6 +126,169 @@ class OrganismController {
             , @RestApiParam(name = "publicMode", type = "boolean", paramType = RestApiParamType.QUERY, description = "a flag for whether the organism appears as in the public genomes list")
             , @RestApiParam(name = "commonName", type = "string", paramType = RestApiParamType.QUERY, description = "a name used for the organism")
             , @RestApiParam(name = "metadata", type = "string", paramType = RestApiParamType.QUERY, description = "organism metadata")
+            , @RestApiParam(name = "sequenceData", type = "file", paramType = RestApiParamType.QUERY, description = "organism metadata")
+    ])
+    @Transactional
+    def addOrganismWithSequence() {
+        JSONObject organismJson = permissionService.handleInput(request, params)
+        String clientToken = organismJson.getString(FeatureStringEnum.CLIENT_TOKEN.value)
+        try {
+            if (permissionService.isUserAdmin(permissionService.getCurrentUser(organismJson))) {
+                if (organismJson.get("commonName") == "" || organismJson.get("directory") == "") {
+                    throw new Exception('empty fields detected')
+                }
+
+                log.debug "Adding ${organismJson.publicMode}"
+                Organism organism = new Organism(
+                        commonName: organismJson.commonName
+                        , directory: organismJson.directory
+                        , blatdb: organismJson.blatdb
+                        , species: organismJson.species
+                        , genus: organismJson.genus
+                        , metadata: organismJson.metadata
+                        , publicMode: organismJson.publicMode
+                ).save(failOnError: true, flush: true, insert: true)
+                log.debug "organism ${organism as JSON}"
+
+                preferenceService.setCurrentOrganism(permissionService.getCurrentUser(organismJson), organism, clientToken)
+
+                // send file using:
+//            curl \
+                //  -F "userid=1" \
+                //  -F "filecomment=This is an image file" \
+                //  -F "sequenceData=@/home/user1/Desktop/jbrowse/sample/seq.zip" \
+                //  localhost:8080/apollo
+                byte[] buffer = new byte[1024];
+
+                try {
+                    File file = request.getFile("sequenceData)")
+                    File folder = new File(organism.directory);
```

Comment:
This requires the user to have knowledge of where data should go on the apollo server, and will open up a LOT of problems.

Since this will be a separate API from "addOrganism", you could remove the "directory" parameter completely. I'd recommend replacing it with something like the following (pseudocode)

```
new File(apollo.base_organism_directory + File.separator +  request.user.id + File.separator + uuid.generate())
```

which would be much safer and decrease user control over where files can be written.